### PR TITLE
Clarify documentation of inventory actions

### PR DESCRIPTION
**Pull request by Wuzzy2**
This PR improves the help for `minetest.register_allow_player_inventory_action` and `minetest.register_on_player_inventory_action` in `lua_api.txt`.

The text was not clear enough on what these do *exactly*. It was written in a way that sounded like this allows you to capture *ALL* inventory changes of the player inventory. However, this is not true: It only captures inventory changes within the player inventory or when the item was exchanged with another inventory. If an item was added with e.g. `inv:set_stack` or `inv:add_item` or similar, this callback is *NOT* called. This has tripped me up in my modding.

Thus, I explained what the 3 actions "put", "take" and "move" mean so it should be clear that only those 3 events matter (and not e.g. a `set_stack` call)

I also flipped the order in which these functions appear in the documentation simply to make it a bit easier to explain.

### CODE REVIEW

**Review by SmallJoker — APPROVED** [review]
LGTM

File: doc/lua_api.txt [hunk 1]
```diff
@@ -5317,22 +5317,26 @@ Call these functions only at load time!
...
         * `move`: `{from_list=string, to_list=string, from_index=number, to_index=number, count=number}`
         * `put`:  `{listname=string, index=number, stack=ItemStack}`
         * `take`: Same as `put`
+    * Does not accept or handle any return value.
+* `minetest.register_allow_player_inventory_action(function(player, action, inventory, inventory_info))`
+    * Determines how much of a stack may be taken, put or moved to a
+      player inventory.
+    * Function arguments: see `minetest.register_on_player_inventory_action`
     * Return a numeric value to limit the amount of items to be taken, put or
       moved. A value of `-1` for `take` will make the source stack infinite.
```
Comment (Desour) [inline doc/lua_api.txt:5339]: This is called before the `register_on_player_inventory_action`, right?
Does returning 0 prevent the `register_on_player_inventory_action` callback? If yes, please add a note.

**Comment by Desour** [comment]
> Hmmm, I actually don't konw if returning 0 does something special. :-(

Seems so at least:
https://github.com/minetest/minetest/blob/b7359f5fa93f531986c86b1ced20bb4584593b0a/src/inventorymanager.cpp#L442-L443
Haven't tested though.
Also, it looks like no items are moved if there would be a swap but only a part of the stack is allowed.

**Comment by Wuzzy2** [comment]
I don't know how to proceed. I think this PR is correct.

**Comment by Desour** [comment]
Sorry, I shouldn't have requested you to expand the PR.

For consistency with this:
https://github.com/minetest/minetest/blob/38e005294f85cee2f7a9571b4a1b20154a6b305a/doc/lua_api.txt#L6808
, could you please restore the old order of the callbacks (i.e. `allow_` before `on_`)?
It's fine then.

**Comment by Wuzzy2** [comment]
Done.

**Review by Desour — APPROVED** [review]

**Pull request merged**